base/dcp_sharded.go

```diff
@@ -337,6 +337,16 @@ func initCBGTManager(ctx context.Context, bucket Bucket, spec BucketSpec, cfgSG
 		sourceUUID:    bucketUUID,
 	}
 
+	// Define rollback hook to be called by cbgt during rollback. This is used to kick the janitor to reassign feeds
+	// after a full rollback of a pindex
+	cbgt.RollbackHook = func(phase cbgt.RollbackPhase, pindex *cbgt.PIndex) (err error) {
```

Comment:
Since this modifies a _global_ in cbgt, can we move this to `init` where we add the other globals? https://github.com/couchbase/sync_gateway/blob/a68b162bba09dddbd333e2c4fda733f36d91488a/base/dcp_feed_type.go#L119

Reply:
I agree this makes sense but we need access to the manager to kick the janitor, which doesn't look like it can be done there?

Reply:
Then we need to figure out how to run a function that passes in a manager for a specific bucket `initCBGTManager` is called once per bucket and then a second database would overwrite this callback for the first.

I think we might be able to do something with a wrapper function to look up mgr based on pindex and store this globally, or we could ask cbgt team to change their API so this isn't required? I don't know if they have one manager per instance, but that seems wacky.

Reply:
Abandoned this approach for fix in cbgt.